lib/screens/widgets/location_input.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
+
+class LocationInput extends StatefulWidget {
+  final Function(String) onLocationChanged;
+
+  const LocationInput({super.key, required this.onLocationChanged});
+
+  @override
+  _LocationInputState createState() => _LocationInputState();
```

Comment:
To Avoid flutter warning. You should install flutter snippets extension for easy creation of stateful widgets
```suggestion
  State<LocationInput> createState() => _LocationInputState();
```